libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/DrillTargets/DrillTargetAttributeHierarchyItem.tsx

```diff
@@ -1,29 +1,126 @@
-import { selectCatalogAttributeHierarchies, useDashboardSelector } from "../../../../../model/index.js";
+import {
+    selectIgnoredDrillDownHierarchiesByWidgetRef,
+    selectCatalogAttributeHierarchies,
+    useDashboardSelector,
+} from "../../../../../model/index.js";
+import { useIntl } from "react-intl";
+import isEmpty from "lodash/isEmpty.js";
+import { AttributeHierarchyList } from "./AttributeHierarchyList.js";
 
 interface IDrillTargetDashboardItemProps {
     config: IDrillDownAttributeHierarchyConfig;
+    onSelect: (targetItem: ICatalogAttributeHierarchy) => void;
 }
 
-const DrillTargetAttributeHierarchyItem: React.FC<IDrillTargetDashboardItemProps> = ({ config }) => {
+const DROPDOWN_ALIGN_POINTS = [
+    {
+        align: "bl tl",
+        offset: {
+            x: 0,
+            y: 4,
+        },
+    },
+    {
+        align: "tl bl",
+        offset: {
+            x: 0,
+            y: -4,
+        },
+    },
+];
+
+const DrillTargetAttributeHierarchyItem: React.FC<IDrillTargetDashboardItemProps> = ({
+    config,
+    onSelect,
+}) => {
+    const intl = useIntl();
     const catalogAttributeHierarchies = useDashboardSelector(selectCatalogAttributeHierarchies);
-    const selectedCatalogAttributeHierarchy = catalogAttributeHierarchies.find(
-        (it) => objRefToString(it.attributeHierarchy.ref) === objRefToString(config.attributeHierarchyRef),
+    const ignoredDrillDownHierarchies = useDashboardSelector(
+        selectIgnoredDrillDownHierarchiesByWidgetRef(config.widgetRef),
+    );
+
+    const attributeDescriptor = config.attributes.find(
+        (attr) => attr.attributeHeader.localIdentifier === config.originLocalIdentifier,
     );
 
+    const selectedCatalogAttributeHierarchy = config.complete
+        ? catalogAttributeHierarchies.find((hierarchy) =>
+              areObjRefsEqual(hierarchy.attributeHierarchy.ref, config.attributeHierarchyRef),
+          )
+        : null;
+
+    const existBlacklistHierarchies = catalogAttributeHierarchies.filter((hierarchy) => {
+        return ignoredDrillDownHierarchies.some((ref) => {
+            if (isAttributeHierarchyReference(ref)) {
+                return (
+                    areObjRefsEqual(ref.attributeHierarchy, hierarchy.attributeHierarchy.ref) &&
+                    objRefToString(ref.label) === attributeDescriptor?.attributeHeader.identifier
+                );
+            } else {
+                return (
+                    areObjRefsEqual(ref.dateHierarchyTemplate, hierarchy.attributeHierarchy.ref) &&
+                    objRefToString(ref.dateDatasetAttribute) ===
+                        attributeDescriptor?.attributeHeader.identifier
+                );
+            }
+        });
+    });
+
+    const emptyHierarchyMessage = intl.formatMessage(messages.emtpyHierarchyInfo);
+    const shouldShowEmptyMessage = isEmpty(existBlacklistHierarchies) && !config.complete;
```

Comment:
In cases where the origin attribute exists in more than one attribute hierarchy, displaying an empty message does not make sense. I would suggest only showing the empty message if the origin attribute is not included in any attribute hierarchy. In such cases, the hierarchy dropdown should be shown, but the item not in the blacklist will be disabled.

Could you help check, thanks!

Reply:
Hi @xMort, What do you think about this case? It isn't mentioned in mockup.

Reply:
BTW, if we display the empty message in this case, that means the dropdown will not render, and users cannot edit the attribute hierarchy.

cc @phong-nguyen-duy , @xMort

Reply:
I think @khanhhuynh-gooddata is right.

Reply:
Done, the dropdown will show all hierarchies which contains the origin attribute and the items are not in the blacklist will be disabled